mirdata/dataset.py

```diff
@@ -0,0 +1,157 @@
+"""Module containing the Dataset base class
+"""
+import importlib
+import os
+import random
+
+from mirdata import download_utils
+from mirdata import utils
+
+
+class Dataset(object):
+    """mirdata Dataset object
+
+    Usage example:
+    orchset = mirdata.Dataset('orchset')  # get the orchset dataset
+    orchset.download()  # download orchset
+    orchset.validate()  # validate orchset
+    track = orchset.choice()  # load a random track
+    print(track)  # see what data a track contains
+    orchset.track_ids()  # load all track ids
+
+    Attributes:
+        dataset (str): the identifier of the dataset
+        bibtex (str): dataset citation/s in bibtex format
+        remotes (dict): data to be downloaded
+        index (dict): dataset file index
+        download_info (str): download instructions or caveats
+        track_object (mirdata.track.Track): an uninstantiated Track object
+        dataset_dir (str): dataset save folder
+        readme (str): information about the dataset
+        data_home (str): path where mirdata will look for the dataset
+
+    """
+
+    def __init__(self, dataset, data_home=None):
+        """Inits a dataset by name and data location"""
+        module = importlib.import_module("mirdata.{}".format(dataset))
+        self.dataset = dataset
+        self.bibtex = getattr(module, "BIBTEX", "No citation data provided")
+        self.remotes = getattr(module, "REMOTES", {})
+        self.index = module.DATA.index
+        self.download_info = getattr(module, "DOWNLOAD_INFO", None)
+        self.track_object = getattr(module, "Track", None)
+        self.dataset_dir = module.DATASET_DIR
+        self.download_fn = getattr(module, "download", download_utils.downloader)
+        self.readme = module.__doc__
+
+        if data_home is None:
+            self.data_home = self.default_path
+        else:
+            self.data_home = data_home
+
+    @property
+    def default_path(self):
+        """Get the default path for the dataset
+
+        Returns:
+            default_path (str): Local path to the dataset
+        """
+        mir_datasets_dir = os.path.join(os.getenv("HOME", "/tmp"), "mir_datasets")
+        return os.path.join(mir_datasets_dir, self.dataset_dir)
+
+    def track(self, track_id):
+        """Load a track by track_id
+
+        Args:
+            track_id (str): track id of the track
+        
+        Returns:
+            track (dataset.Track): an instance of this dataset's Track object
+        """
+        if self.track_object is None:
+            raise NotImplementedError
+        else:
+            return self.track_object(track_id, self.data_home)
+
+    def load_tracks(self):
+        """Load all tracks in the dataset
+
+        Returns:
+            (dict): {`track_id`: track data}
+        
+        Raises:
+            NotImplementedError: If the dataset does not support Track objects
+        """
+        return {self.track(track_id) for track_id in self.track_ids}
```

Comment:
maybe add the `tqdm` as in [here](https://github.com/mir-dataset-loaders/mirdata/pull/219/files#diff-8d8cb5987f2bdf3f193ef08ea93956bc4d8d6adf147ba19fd588957114fc4fdfR119-R122)

Reply:
i thought about it, but it's not slow to load the tracks, since the objects don't load any data.